sklearn/preprocessing/_function_transformer.py

```diff
@@ -64,11 +64,31 @@ def fit(self, X, y=None):
             check_array(X, self.accept_sparse)
         return self
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length len(input_features), optional
+            String names for input features if available. By default,
+            None is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length len(input_features)
+
+        """
+        if input_features is not None:
```

Comment:
I think this logic is incorrect. `func` is not applied to each feature independently, but to X as a whole. The only way to get sensible feature names out of a `FunctionTransformer` is to have another parameter specify `get_feature_names`, essentially, or else to count the number of columns output from `transform` and make up default names on that basis.